hardening/container/day2-upgrade/test.py

```diff
@@ -0,0 +1,154 @@
+from pathlib import Path
+
+from lib import results, oscap, versions, virt, podman, util
+from conf import remediation
+
+
+virt.Host.setup()
+
+profile = util.get_test_name().rpartition('/')[2]
+oscap_repo = os.environ.get('CONTEST_OSCAP_REPOFILE')
+
+# note that the .wipe() is necessary here, as we are not calling any .install()
+# function that would normally perform it
+guest = virt.Guest()
+guest.wipe()
+guest.generate_ssh_keypair()
+
+# select appropriate container image based on host OS
+major = versions.rhel.major
+minor = versions.rhel.minor
+if versions.rhel.is_true_rhel():
+    src_image = f'images.paas.redhat.com/testingfarm/rhel-bootc:{major}.{minor}'
+else:
+    src_image = f'quay.io/centos-bootc/centos-bootc:stream{major}'
+
+# prepare a RpmPack with testing-specific hacks
+# - copy it to CWD because podman cannot handle absolute paths (or relative ones
+#   going above CWD) as source for COPY or RUN --mount
+pack = util.RpmPack()
+if oscap_repo:
+    pack.add_file(oscap_repo)
+pack.add_sshd_late_start()
+with pack.build() as pack_binrpm:
+    shutil.copy(pack_binrpm, 'contest-pack.rpm')
+
+# prepare a Container file for making a hardened image using the old data stream
+with util.get_old_datastream() as old_ds:
+    oscap.unselect_rules(
+        old_ds, 'remediation-old-ds.xml', remediation.excludes())
+cfile = podman.Containerfile()
+cfile += util.dedent(fr'''
+    FROM {src_image}
+    # install testing-specific RpmPack
+    COPY contest-pack.rpm /root/.
+    RUN dnf -y install /root/contest-pack.rpm && rm -f /root/contest-pack.rpm
+    # copy over testing-specific datastream
+    COPY remediation-old-ds.xml /root/
+    # install and run oscap-im to harden the image
+    RUN dnf -y install openscap-utils
+    RUN oscap-im --profile '{profile}' \
+        --results-arf /root/remediation-old-arf.xml /root/remediation-old-ds.xml
+    RUN bootc container lint
+''')
+cfile.add_ssh_pubkey(guest.ssh_pubkey)
+cfile.write_to('Containerfile')
+
+podman.podman('pull', src_image)
+podman.podman('image', 'build', '--tag', 'contest-hardened-old', '.')
+
+# prepare a Container file for making a hardened image using the new data stream
+oscap.unselect_rules(util.get_datastream(), 'remediation-new-ds.xml', remediation.excludes())
+cfile2 = podman.Containerfile()
+cfile2 += util.dedent(fr'''
+    FROM {src_image}
+    # install testing-specific RpmPack
+    COPY contest-pack.rpm /root/.
+    RUN dnf -y install /root/contest-pack.rpm && rm -f /root/contest-pack.rpm
+    # copy over testing-specific datastream
+    COPY remediation-new-ds.xml /root/
+    # install and run oscap-im to harden the image
+    RUN dnf -y install openscap-utils
+    RUN oscap-im --profile '{profile}' \
+        --results-arf /etc/remediation-new-arf.xml /root/remediation-new-ds.xml
+    RUN bootc container lint
+''')
+cfile2.add_ssh_pubkey(guest.ssh_pubkey)
+cfile2.write_to('Containerfile')
+podman.podman('image', 'build', '--tag', 'contest-hardened-new', '.')
```

Comment:
This seems like unnecessary duplication just for "old" vs "new".

Have a look at how this is done in the other `old-new` test: https://github.com/RHSecurityCompliance/contest/blob/f045e57af1066d276ce24a2da47e51143b893386/hardening/oscap/old-new/test.py#L29-L46

Perhaps this code could also be moved to a function, which would then take filename arguments that could be specified as
```python
    build_image('remediation-old.xml', 'remediation-arf-old.xml', 'remediation-arf-old2.xml')
    build_image('remediation-new.xml', 'remediation-arf-new.xml', 'remediation-arf-new2.xml')
```
or something like that?